src/ThemeProvider.tsx

```diff
@@ -108,11 +108,11 @@ function useSystemColorMode() {
     }
 
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    media?.addEventListener('change', handleChange)
+    media?.addEventListener?.('change', handleChange)
```

Comment:
Can we use [`.addListener()`](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener) when `.addEventListener()` is not supported?